doc/dev/custom_devices.rst

```diff
@@ -17,91 +17,116 @@ Additionally, all measurements will always return ``true``.
             {
             }
 
-            virtual auto Expval(ObsIdType) -> double { return 0.0; }
-            virtual auto Var(ObsIdType) -> double { return 0.0; }
-            virtual void State(DataView<std::complex<double>, 1> &) {}
-            virtual void Probs(DataView<double, 1> &) {}
-            virtual void PartialProbs(DataView<double, 1> &, const std::vector<QubitIdType> &) {}
-            virtual void Sample(DataView<double, 2> &, size_t) {}
-            virtual void PartialSample(DataView<double, 2> &, const std::vector<QubitIdType> &, size_t) {}
-            virtual void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) {}
-
-            virtual void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
-                                       const std::vector<QubitIdType> &, size_t)
+            auto Expval(ObsIdType) -> double override { return 0.0; }
+            auto Var(ObsIdType) -> double override { return 0.0; }
+            void State(DataView<std::complex<double>, 1> &) override {}
+            void Probs(DataView<double, 1> &) override {}
+            void PartialProbs(DataView<double, 1> &, const std::vector<QubitIdType> &) override {}
+            void Sample(DataView<double, 2> &, size_t) override {}
+            void PartialSample(DataView<double, 2> &, const std::vector<QubitIdType> &, size_t) override {}
+            void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) override {}
+
+            void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
+                                    const std::vector<QubitIdType> &, size_t) override
             {
             }
 
-            virtual void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) {}
+            void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) override {}
         };
 
-In addition to implementing the ``QuantumDevice`` class, one must implement the following method:
+In addition to implementing the ``QuantumDevice`` class, one must implement the following method
+with the name ``CustomDevice`` + ``Factory``:
 
 .. code-block:: c++
 
     extern "C" Catalyst::Runtime::QuantumDevice*
-    getCustomDevice() { return new CustomDevice(); }
+    CustomDeviceFactory(bool status, const std::string &kwargs)
+    {
+        return new CustomDevice(status, kwargs); }
+
+The use of the term ``Factory`` in ``CustomDeviceFactory`` is a runtime convention that combines the name
+of the device class, in this case, ``CustomDevice``, with the suffix ``Factory``. This signifies that the
+primary purpose of the method is to act as a factory for creating instances of the ``CustomDevice`` class.
+
+.. warning::
+
+    The runtime exclusively supports factory methods with the name of the device class followed by ``Factory``.
+    Please ensure that your method adheres to this naming convention.
```

Comment:
```suggestion
Importantly, the ``<DeviceIdentifier>`` string in the entry point function needs to match
exactly what is supplied to the ``__quantum__rt__device("rtd_name", "<DeviceIdentifier>")``
runtime instruction in compiled user programs, or what is returned from the ``get_c_interface``
function when integrating the device into a PennyLane plugin (see section further down).
```

Reply:
@dime10 curious what your reason is here for removing the `.. warning::` admonition? No strong feelings either way, but I often find that admonitions, if not overused, allow important information to 'stand out' better. Especially in reference documentation, that is often skim read

Reply:
Oh I didn't mean for the change to specifically remove it, I just rewrote the instructions into flowing text as follows:

> In addition to implementing the ``QuantumDevice`` class, one must implement an entry point for the
> device library with the name ``<DeviceIdentifier>Factory``, where ``<DeviceIdentifier>`` is used to
> unique the entry point symbol. As an example, we use the identifier ``CustomDevice``:
>
> ```py
> extern "C" Catalyst::Runtime::QuantumDevice*
> CustomDeviceFactory(bool status, const std::string &kwargs) {
>    return new CustomDevice(status, kwargs);
> }
> ```
>
> The entry point function acts as a factory method for the device class. Note that a plugin library
> may also provide several factory methods in case it packages multiple devices into the same
> library. However, it is important that the device identifier be unique, as best as possible, to
> avoid clashes with other plugins. 
>
> Importantly, the ``<DeviceIdentifier>`` string in the entry point function needs to match
> exactly what is supplied to the ``__quantum__rt__device("rtd_name", "<DeviceIdentifier>")``
> runtime instruction in compiled user programs, or what is returned from the ``get_c_interface``
> function when integrating the device into a PennyLane plugin (see section further down).
>
> Additionally, the entry point function must accept two arguments defined as below, which
> can be forwarded to the device constructor (e.g. ``CustomDevice(status, kwargs)``) if
> appropriate:
> - ``status``: A boolean to control the initial value of the tape recording system in the runtime.
>  If ``true``, this will cache the entire operations and observables in a program.
>  This may be used to compute gradient of a quantum circuit with taking advantage of gradient
>  methods provided by backend devices if supported.
> - ``kwargs``: A string of device specifications and options, represented in Python dictionary format.
>  This may contain information such as device type and a number of shots:
>  ``{'device_type': 'lightning.qubit', 'shots': 0}``.

All of the parts are equally important I think if you want to implement a custom device, but let me know if you think a warning box would still make sense somewhere :)

Reply:
all good! Was just curious if there was an dislike-of-admonitions reason for it lol